samples/client/petstore/swift5/urlsessionLibrary/PetstoreClient/Classes/OpenAPIs/Models/Order.swift

```diff
@@ -30,5 +31,61 @@ public struct Order: Codable {
         self.status = status
         self.complete = complete
     }
+    public enum CodingKeys: String, CodingKey, CaseIterable {
+        case id
+        case petId
+        case quantity
+        case shipDate
+        case status
+        case complete
+    }
+
+    // Encodable protocol methods
+
+    public func encode(to encoder: Encoder) throws {
```

Comment:
@aymanbagabas this should only be done when using classes, not structs.

Reply:
[Codable](https://developer.apple.com/documentation/swift/codable) can be applied to both AFAICT. I added `encode(encoder:)` and `init(decoder:)` explicitly to decode null values since you _need_ to use `encoderIfPresent()` and `decoderIfPresent()` to capture nulls instead of ignoring the optionals all together.

Reply:
I know we can use explicitly `encode(encoder:)` to `structs`, but I think we should rely on the implicit implementation, because it's less code to maintain and one less source of bugs.

Reply:
As far as I now, the implicit implementation of `encode(encoder:)` already deals with the nulls, or not?

Reply:
I believe you're right. I meant to encode nullable variables. To do so, you gotta use `encode(encoder:)` instead of `encodeIfPresent(encoder:)` for optional values so that it encodes nulls. We really don't need `init(decoder:)` so I'm just gonna drop that one. Here's a good example of what I'm talking about https://stackoverflow.com/a/47268112/10913628.

These are the possible four cases of encoding. I'm using String as an example.

| isNullable | isRequired | type | func
|-----------|------------|------|-------|
| true | true | String? | encode() |
| true | false | String? | encodeIfPresent() |
| false | true | String | encode() |
| false | false | String? | encodeIfPresent() |

My question is, does the generator always has `isNullable` and `isRequired` available? Because I know that `nullable` is an OAI 3.0 feature and it doesn't exist in either 2.0 nor 3.1.

Reply:
Since that's a question related to OAI, I don't know the answer.
But I think everything comes down to `String?` or `String`.
Looking at the first case, `isNullable` `true`, and `isRequired` `true`, I'm not sure if the `func` should be `encode()` or  `encodeIfPresent()`.

Reply:
If we want to encode `null` in the generated JSON, we have to use `encode()` instead of `encodeIfPresent()`. Note that `encodeIfPresent()` is the one used for optionals and that's the default implementation of `func encode(to:)`.

Reply:
@4brunu I will make this change to encode `null`s in the encoded JSONs. I'm also gonna start splitting this PR into smaller ones.

Reply:
Thanks for your effort in splitting this big PR into small ones, that will make it a lot easier to review each PR.

Some good candidates to a small PR are this one, Any to JSON, reserved word _ or `.

Ideally this PR end's up just having the Vapor changes.